### PR TITLE
[cli] Fix external rewrites/redirects in dev

### PR DESCRIPTION
**Pull request by MaxLeiter**
You can locally test rewriting with https://github.com/Jakst/vercel-next-rewrites

https://app.clubhouse.io/vercel/story/6330

Fixes #4538

### CODE REVIEW

File: packages/now-cli/src/util/dev/router.ts [hunk 1]
```diff
@@ -156,7 +165,6 @@ export async function devRouter(
           }
         }
 
-        const isDestUrl = isURL(destPath);
```
Comment (styfle) [inline packages/now-cli/src/util/dev/router.ts:159]: Let's add a dev integration test like the one you linked to in the desc

File: packages/now-cli/test/dev/integration.js [hunk 2]
```diff
@@ -111,7 +111,7 @@ async function exec(directory, args = []) {
 }
 
 async function runNpmInstall(fixturePath) {
-  if (await fs.exists(join(fixturePath, 'package.json'))) {
```
Comment (MaxLeiter) [inline packages/now-cli/test/dev/integration.js:114]: `fs.exists` is deprecated

File: packages/now-cli/test/dev/integration.js [hunk 3]
```diff
@@ -817,6 +817,26 @@ test(
...
+      const vcRobots = `https://vercel.com/robots.txt`;
+      await testPath(200, '/rewrite', /User-Agent: \*/m);
+      await testPath(308, '/redirect', `Redirecting to ${vcRobots} (308)`, {
+        Location: vcRobots,
+      });
+      await testPath(307, '/tempRedirect', `Redirecting to ${vcRobots} (307)`, {
+        Location: vcRobots,
+      });
+
+      t.pass();
```
Comment (TooTallNate) [inline packages/now-cli/test/dev/integration.js]: Is this necessary?

File: packages/now-cli/test/dev/integration.js [hunk 4]
```diff
@@ -817,6 +817,26 @@ test(
...
+    },
+    { skipDeploy: true }
```
Comment (TooTallNate) [inline packages/now-cli/test/dev/integration.js]: Let's test prod too?

**Review by TooTallNate — APPROVED** [review]
You _could_ also add some unit tests to `test/dev-router.unit.js`, but I suppose your test covers it already.

**Review by styfle — APPROVED** [review]
Nice, thanks! 🏅

**Pull request merged**